More Resomi sprites

martinlyra opened the pull request:
This branch "resomi-sprites" purposed for pull, includes new sprites for items, clothing and equipment that were previously did not have sprites when used by Resomi. Using the Human sprites instead which are usually off and obscure when seen on Resomi.

So far this pull request of the branch adds following:

- Modifications to accessory.dm and update_icons.dm to make it possible to use non-human sprite-sheets available for accessories. Layman terms; made accessories more 'magic' and more pleasant for non-human users.
- Adds sprites for equipment used by Resomi:
 - Webbings & Vests
   - Brown webbing vest (Engineering)
    - White webbing vest (Medical)
    - Black webbing vest (Security & Mercenaries)
    - Standard webbing

Images:

![Brown webbing vest](http://i.imgur.com/mFhkQaK.png)
![Brown webbing vest](http://i.imgur.com/eJ3X3cW.png)

![White webbing vest](http://i.imgur.com/fwnQSht.png)
![White webbing vest](http://i.imgur.com/xaR5AyC.png)

![Black webbing vest](http://i.imgur.com/x4WhJDl.png)
![Black webbing vest](http://i.imgur.com/0mV2GNw.png)

![Webbing](http://i.imgur.com/dZ9ACId.png)
![Webbing](http://i.imgur.com/hq0C0uz.png)

mwerezak commented on code/modules/clothing/under/accessories/accessory.dm:
```diff
@@ -12,14 +12,16 @@
 	return ..()
 
-/obj/item/clothing/accessory/proc/get_inv_overlay()
+/obj/item/clothing/accessory/proc/get_inv_overlay(var/mob/user_mob)
```
This is used for the inventory, changing it will look odd as the inventory sprites are not species specific.

martinlyra replied:
Perhaps I could wire it to a function that update both overlays when picked up and equipped on by person of different species?

mwerezak replied:
The only time changing it based on species would make sense is if `sprite_sheets_obj` is being used.

martinlyra replied:
I've fixed it so it no longer relies on the mob overlay sprite, instead it gets the appropriate one from the default sheet. The only flaw so far is when if there is a state that exists in the spritesheet, but not in the default. The inventory overlay will turn up as invisible.

mwerezak replied:
The same is true of every item in the game that uses a spritesheet.

martinlyra commented:
Disregard the dev -> resomi-sprite branch commit, it wasn't supposed to happen.

Anyway, @mwerezak, I've removed reverted some changes to the get_inv_overylay and I'm due to test if it cause problems or not.

PsiOmegaDelta commented:
People whom seem to have an understanding of how the sprite buildup works, is this good to go?

martinlyra commented:
Squashed all the nine commits into a commit worth of 11 commits.
New changes in the last three commits:
- New sprite for Head of Security's coat and trenchcoat
- Fixed some pixels on the Detective's brown coat. Grey coat not yet supported, to my knowledge.

GinjaNinja32 commented on code/modules/clothing/under/accessories/accessory.dm:
```diff
@@ -54,7 +65,7 @@
 	if(!has_suit)
 		return
-	has_suit.overlays -= get_inv_overlay()
+	has_suit.overlays -= get_inv_overlay(0)
```
use `null` here if you want to be explicit about passing no parameter, not 0; it's a reference, not a number.

martinlyra commented:
Said and done, @GinjaNinja32 

But please postpone the merging until I give a green light. I feel like adding more sprites.

martinlyra commented:
Now I've finished the first batch, now it should be ready for merging.

The new changes were:
 - Added sprite for the grey detective coat for Resomi
 - Added sprites for the old and new armor and webbing vests (suit-slot ones) - all of them (except the high-tech stuff, like reflective and etc.)

mwerezak commented:
This doesn't work if a non-resomi wears the webbing first, due to the overlay caching.

![screenshot-space station 13-2](https://cloud.githubusercontent.com/assets/242428/15025606/0c10911e-1208-11e6-8491-ee7f99bde7ff.png)

martinlyra commented:
Any help would be appreciated.

martinlyra commented:
@mwerezak Apparently from looks of it, I believe I've fixed your mentioned issue by removing an expression from get_mob_overlay function. Now it will always replace the overlay every time the function is called.

mwerezak commented:
Getting rid of caching altogether would fix it but that's not ideal.

It shouldn't be too hard to add caching for alternate sprite sheets, so I'd rather not merge this until that is included.

mwerezak commented:
Maybe you could store the overlays in an assoc list keyed by user_human.species.get_bodytype()?

martinlyra commented:
How is this implementation?

```
	var/image/mob_overlay_override = null
	...
	var/list/mob_overlays = list()
...

/obj/item/clothing/accessory/proc/get_mob_overlay(var/mob/user_mob)
	var/tmp_icon_state = "[overlay_state? "[overlay_state]" : "[icon_state]"]"
	
	if (!icon_override)
		if(!mob_overlays.len) // cache the icons
			mob_overlays["default"] = image("icon" = INV_ACCESSORIES_DEF_ICON, "icon_state" = "[tmp_icon_state]")
			for (var/entry in sprite_sheets)
				mob_overlays[entry] = image("icon" = sprite_sheets[entry], "icon_state" = "[tmp_icon_state]")
			
		var/user_mob_species = "Human"
		if(user_mob && istype(user_mob, /mob/living/carbon/human))
			var/mob/living/carbon/human/user_human = user_mob
			user_mob_species = user_human.species.get_bodytype()
		if (sprite_sheets[DETERMINATE(user_mob_species)] && (tmp_icon_state in icon_states(sprite_sheets[DETERMINATE(user_mob_species)])))
			mob_overlay = mob_overlays[user_mob_species]
		else
			mob_overlay = mob_overlays["default"]
	else
		if (!mob_overlay_override)
			if("[tmp_icon_state]_mob" in icon_states(icon_override))
				tmp_icon_state = "[tmp_icon_state]_mob"
			mob_overlay_override = image("icon" = icon_override, "icon_state" = "[tmp_icon_state]")
		mob_overlay = mob_overlay_override
	return mob_overlay
```
The only problem with this they won't update according if the icon_state has been changed since the cache. Unless we add an argument variable for caching the sprites again.

mwerezak commented:
I dunno, why not just
```
overlay = overlay_cache[user_human.species.get_bodytype()]
if(!overlay)
    //create the new overlay
```

If you're willing to wait a little longer I'll write it myself and PR it to you.

mwerezak commented:
@martinlyra https://github.com/martinlyra/Baystation12/pull/1

martinlyra commented:
Looks like Git has messed up various of the previous commits when rebasing to the latest dev branch.

mwerezak commented:
Maybe try `git reset --hard f019279e8c4399307d93986b0a5637015c237ac1` followed by a `git merge upstream/dev`?

martinlyra commented:
The merge conflict has (hopefully) been resolved in the merge commit.

mwerezak commented:
I can't sign off on this now seeing as it has my own code, so just a reminder this still needs to be reviewed,
